component/discovery/transformer/types.go

```diff
@@ -0,0 +1,107 @@
+package transformer
+
+import (
+	"fmt"
+
+	"github.com/grafana/regexp"
+)
+
+// Action is the relabelling action to be performed.
+type Action string
+
+// Regexp encapsulates the Regexp type from Grafana's
+// fork of the Go stdlib regexp package.
+// TODO (@tpaschalis) This encapsulation already exists in Prometheus' relabel.Regexp
+// so not sure whether to also move it here for now.
+type Regexp struct {
+	re       *regexp.Regexp
+	original string
+}
```

Comment:
I think this can be `type Regexp *regexp.Regexp`. It's also not necessary to keep the original string; you can just call [`regexp/Regexp.String`](https://pkg.go.dev/regexp#Regexp.String) if you need the original input.

Reply:
I tried using `type Regexp *regexp.Regexp` but turns out we can't define methods on pointer types.

We can either use `type Regexp regexp.Regexp` (without the pointer), which involves some back-and-forth casting or just embedding it in a struct a-la
```
type Regexp struct {
	*regexp.Regexp
}
```

Does that look okay to you?

Reply:
Either one is fine IMO, but embedding a pointer might be potentially more prone to developer error.

Reply:
Also, this can be improved upstream, one less string to keep around :) I've opened https://github.com/prometheus/prometheus/pull/10846, I'll check if we can generate a benchmark for it.